### DISCUSSION TITLE
widget area

### DISCUSSION THREAD

**Discussion by bobcolo** [EdgeTX/edgetx#2780] [Ideas]
hello
 why screen area widget is limited wrt opentx 
less possibilities for screen display , I have several widget that I can't used with EDGETX for example I can't manage various brackground screen on each model

thanks

**Comment by pfeerick** [comment]
I'm not sure what you mean... you have a lot of control over the widget area/size - you can change the layout to quite a few different grid types, and can also turn off things like the sliders, trims, etc in order to make the widget area bigger or smaller. Perhaps give some specific examples?

![image](https://user-images.githubusercontent.com/5500713/203264909-6d212382-7965-4f54-b301-9c82844802fd.png)

**Reply by bobcolo** [reply]
in fact on EDGETX is impossible to print outof the frame you have chosen for the widget this is different from OPENTX 

for example on the same widget I can print 
BAT end RSSI on the frame chosed for widget
model name top right of the screen 
dedicated background for each model


 
![image](https://user-images.githubusercontent.com/118561852/203306552-0205dcc3-bfad-4183-89b7-16f948265bff.png)

**Reply by pfeerick** [reply]
How did you have a different image per model on OpenTX? I only ever saw reference to [it being system-wide](http://www.multirotorguide.com/guide/custom-splash-screen-background-and-model-images-icons-in-opentx-radio/)  ... which can still be done now. 

Widgets being able to draw outside of allocated area is a bug, not a feature - they should only ever be able to draw inside the area they have been allocated. What may change in the future is being able to set how many top bar zones there are, thus allowing for wider zones.

**Comment by bobcolo** [comment]
May be is à bug but it,s useful for me 
i hop they will not fix it in the future
to have background change you have only to draw bitmaps 480 272 
it works perfectly

**Reply by pfeerick** [reply]
> to have background change you have only to draw bitmaps 480 272
it works perfectly

Draw how? Where is the option in OpenTX to do this?

One of the planned future features for EdgeTX is to have model specific backgrounds and themes. #1681

**Reply by bobcolo** [reply]
the widget is installed in topbar

Bitmaps  = Bitmap.open("/WIDGETS/BACKGROUND/"..backgroundimage..".png")
lcd.drawBitmap(Bitmaps ,  0, 45)

it's not exactly a background image but have the same behaviour  when topbar is active

**Reply by pfeerick** [reply]
Unfortunately, that cannot be allowed to work here. Just like with any "out-of-bounds" memory bug, a widget should not draw outside of its defined work area. 

So does that mean you have multiple widgets, or a widget that accepts a parameter, so you can change it on a model by model basis? As you can change the default background by changing the image at `/THEMES/EdgeTX/background.png` (as mentioned in #1681). 

An alternative implementation would be to have an API call that Lua could use to change the background, and let the system deal with the rest.

**Reply by bobcolo** [reply]
in my widget" background" is a integer picture are stored as 1.png, 2.png, 3.png ....etc so i can manage backgound easily by the widget setting -1 -> rdm    0-> defaut   1 to n  ->  picture for the modèle
